### ISSUE TITLE
Upgrade to support library v28.0.0-alpha1

### ISSUE THREAD

**Issue by Gericop** [Gericop/Android-Support-Preference-V7-Fix#146]

**Comment by tcqq** [comment]
@Gericop I have now upgraded my project to Android X, but the code has got an error
`import androidx.preference.Preference;`

Can now use `compile 'com.takisoft.fix:preference-v7:27.1.1.1-alpha1'` version?

**Comment by Gericop** [comment]
It's gonna be `com.takisoft.fix:preference:1.0.0.0-alpha1` but it's not refactored yet so you cannot use it right now.

**Comment by tcqq** [comment]
When can I use `com.takisoft.fix:preference:1.0.0.0-alpha1`?

**Comment by Gericop** [comment]
In the next few days probably.

**Comment by Gericop** [comment]
Sorry, I had a lot to do recently. It's gonna be out only next Tuesday (May 22).

**Comment by tcqq** [comment]
Ok.

**Comment by b95505017** [comment]
Any update?

**Comment by Gericop** [comment]
It's work in progress. This week has been a bit busy unfortunately.

**Comment by Gericop** [comment]
So, I've finally started working on the refactor (first the DateTimePicker and ColorPicker sublibs). Updates coming soon in a new repo.

**Comment by Gericop** [comment]
Work in progress now.

**Comment by Gericop** [comment]
The AndroidX version is available now. Check out https://bintray.com/takisoft/android for the available modules.

Note that it has some breaking changes, like the missing `PreferenceFragmentCompatDividers` class (just use `PreferenceFragmentCompat` from `com.takisoft.preferencex`). Also the license is Apache v2 from now on.

**Comment by Mygod** [comment]
I think we can close this for now.

**Comment by Gericop** [comment]
I'll leave this open until the lib's AndroidX version is released to the master branch and the repo is transferred to its new place. Just a few more days (probably Monday).

**Comment by divadsn** [comment]
any update on that? AndroidX has now a rc version.

**Comment by Gericop** [comment]
@divadsn Not much. You can use the AndroidX version, however, from the link mentioned above (https://bintray.com/takisoft/android) by adding these to your gradle file:

```gradle
implementation 'androidx.preference:preference:1.0.0-rc02'
implementation 'com.takisoft.preferencex:preferencex:1.0.0-alpha2'
```

**Comment by RocketRider** [comment]
The AndroidX 1.0.0 and the 28.0.0 are now stable. Looking forward to upgrade my app ;)

**Comment by Gericop** [comment]
Yeah, probably will publish the final version of the support lib (28.0.0.0) on Monday and AndroidX 1.0.0 the same or the next day.

**Comment by gregkorossy** [comment]
@RocketRider v28.0.0.0 is out. AndroidX has to wait til tomorrow.

**Comment by gregkorossy** [comment]
The AndroidX version is basically ready to roll, however, I need to wait until the two external packages (ColorPicker and DateTimePicker) are available on jcenter.

**Comment by gregkorossy** [comment]
**1.0.0** is available on jcenter. Thanks for the patience, and everyone who contributed, especially @Mygod.

Use this to add it to your project:
```gradle
implementation 'androidx.preference:preference:1.0.0'
implementation 'com.takisoft.preferencex:preferencex:1.0.0'
```

> P.S.: This repo will be "moved" in the coming days to a new one where the AndroidX version will reside while this one becomes dormant.

**Comment by andreluizreis** [comment]
@Gericop I updated the library as you recommended previously. However, Android Studio is not finding the @style/PreferenceFixTheme on my style.xml file. Do you know about this problem? 
`@style/PreferenceFixTheme.Light.DarkActionBar not found.`

**Comment by gregkorossy** [comment]
@andreluizreis That theme is no longer needed, thus it was removed. Just use a regular `AppCompat` or `MaterialComponents` theme.

**Comment by andreluizreis** [comment]
@Gericop , I changed the theme to `Theme.AppCompat.Light.DarkActionBar`. I am using the two imports as you commented above: 
`implementation 'androidx.preference:preference:1.0.0'
implementation 'com.takisoft.preferencex:preferencex:1.0.0'`

I continue to use the `onCreatePreferencesFix` method from `PreferenceFragmentCompat` class from your library. I am receiving the error:
`java.lang.NullPointerException: Attempt to invoke virtual method 'int android.view.View.getPaddingStart()' on a null object reference`.

Could you please give an explanation with the differences between the new library and the current rep. documentation?

**Comment by gregkorossy** [comment]
@andreluizreis Could you open a separate issue with the stack trace and some details, like the version of Android? I cannot reproduce the issue.

**Comment by andreluizreis** [comment]
Yes. I will do tomorrow. Thanks for helping me.

**Comment by andreluizreis** [comment]
@Gericop My app is not giving that error anymore. I can run it now without crash but I don't see any difference with your fix and without it. The items are showing with the wrong indentation. 

Title
 ------> Link Title
 ------> Subtitle
---------------------------
Title 
 ------> Link Title
 ------> Subtitle
---------------------------

Or in my other page

Title Big
---------------------------
Title
  ------> Switch
   ------> Switch
   ------> Switch
---------------------------`

Before the update to androidx, I didn't have problems with these tabs at the front of the items. 
I don't know if some else is getting this same issue. 
Do you think it is better to open a new issue here in the GitHub?

**Comment by gregkorossy** [comment]
@andreluizreis Are you talking about the icon space before the preferences (marked with red rectangle in the picture below)? 
![kep](https://user-images.githubusercontent.com/5181621/46549954-ef1b5280-c8d3-11e8-8467-5263f5cca0c5.png)

**Comment by andreluizreis** [comment]
@Gericop, I am adding an image now. There is a new line below the first title that didn't have before. And the space before the switch. 

![screenshot_2018-10-05-13-23-54](https://user-images.githubusercontent.com/8194164/46550164-3983db80-c8a2-11e8-8276-1b2d17e36aea.png)

**Comment by gregkorossy** [comment]
@andreluizreis Open a new issue and post your XML file as well as this image. Your category elements seem weird.

**Comment by andreluizreis** [comment]
@Gericop Ok. I posted now.
